### ISSUE TITLE
Milestones and Events tabs disabled for this model

### ISSUE THREAD

**Issue by madhurrya** [sartography/spiff-arena#1240]
For the normal user for the instances of this model, the Milestones and Events tabs are disabled. Not sure why. For the admin they are enabled.

https://dev.app.spiff.status.im/process-models/manage-finance:accounts-payable:process-expense-reports:upload-expense-evidence-travel

e.g. https://dev.app.spiff.status.im/i/1023

![Image](https://github.com/sartography/spiff-arena/assets/110527567/431ed335-afc4-43b4-9848-e451fbd207eb)

**Comment by madhurrya** [comment]
Same for this model - https://test.app.spiff.status.im/process-models/manage-finance:accounts-payable:process-expense-reports:upload-expense-evidence-procurement
![image](https://github.com/sartography/spiff-arena/assets/110527567/8d671812-65ae-40ff-947f-be7ce9e64b9c)

**Comment by jasquat** [comment]
@madhurrya what use are you signed in as when you see this? They may not have access to the logs api.

**Comment by madhurrya** [comment]
Normal users like core.contributor@status.im, desktop.project-lead@status.im.
But when they go to other processes they have access to these tabs. Issue is with the upload expense processes.

**Comment by jasquat** [comment]
@madhurrya yeah it looks like internal has access to start those process models but not read the logs in the instances. Those permissions could probably be added to the "start" macro. I'm not sure if that's the correct thing to do but it does make sense that if a user can start a process model then they can also view the logs.

**Comment by madhurrya** [comment]
Oh, I think that's because these 2 processes are started by a message event. I think since it's started by a message, it shouldn't come in 'Start new' and that's why these users don't have access to start it.

But ass you can see in the meta data, it is Started by this user.

![image](https://github.com/sartography/spiff-arena/assets/110527567/37a177ab-dae4-4f91-a368-e7e5f86dbc88)

Also the other user, the budget owner who reviews this process also should be able to see these tabs like in the other models.

I understand what you have said, but I think we'll have to something about this.

**Comment by sashayar13** [comment]
@jasquat I think it should be configured the way so that if there is a user logged as the initiator ("Started by") of a process instance , they should be able to see the Logs.

@harmeet-status fyi

**Comment by madhurrya** [comment]
@sashayar13 I think other approvers of the instance also should be able to see those tabs like it currently works for PP1/PP2.

**Comment by sashayar13** [comment]
Yes @madhurrya , this is a good point

**Comment by jasquat** [comment]
So the simple way to do this is to give the desired group read access to:
* `/logs/[process_model_identifier]`
* `/logs/typeahead-filter-values/[process_model_identifier]`

We currently do not have a permission that would allow a user to see the logs for a process instance associated with them. We could add a `for-me` endpoint to both of those above but that would require dev work.

**Comment by sashayar13** [comment]
@jasquatIs it safe to give permission to the 'internal' group, considering it's the only group that includes requestors and Budget Owners? Are there any potential security risks?

**Comment by jasquat** [comment]
@sashayar13 it seems fine to me. Any user with a status-im email address goes into the internal group so it's up to you and @harmeet-status.

I think from the `/logs/*` endpoint all you can get are the event details like process instance id, task identifier, errors, etc. You can't get task data so giving the internal group access to the full `/logs/*` permission may be fine. You also can't navigate to a full log list from the web-ui, you can only access them from the process instance show page which its own permissions.

**Comment by sashayar13** [comment]
@harmeet-status let me know your thoughts on this

**Comment by harmeet-status** [comment]
I dont think that defining permissions like this is a good way of doing this. Isn't there a way that a process model which has been started by a parent process, can inherit the permissions from that parent?

**Comment by madhurrya** [comment]
@calexh-sar what are we going to do to this issue?
